capabilities/index/index_test.go

```diff
@@ -0,0 +1,37 @@
+package index_test
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/ipfs/go-cid"
+	"github.com/ipld/go-ipld-prime/codec/dagjson"
+	cidlink "github.com/ipld/go-ipld-prime/linking/cid"
+	basicnode "github.com/ipld/go-ipld-prime/node/basic"
+	"github.com/storacha/go-libstoracha/capabilities/index"
+	"github.com/stretchr/testify/require"
+)
+
+func TestRoundTripAddCaveats(t *testing.T) {
+	testCid, err := cid.Parse("bafybeiduiecxoeiqs3gyc6r7v3lymmhserldnpw62qjnhmqsulqjxjmtzi")
+	require.NoError(t, err)
+
+	nb := index.AddCaveats{
+		Index: cidlink.Link{Cid: testCid},
+	}
+
+	node, err := nb.ToIPLD()
+	require.NoError(t, err)
+
+	buf := bytes.NewBuffer([]byte{})
```

Comment:
you don't need `bytes.NewBuffer` to create a new buffer (intuitive, I know).

This is usually achieved just by declaring the variable as `var buf bytes.Buffer`. This works because the zero value for the struct is ready to use, no need for any other preparation.

If you do that, note that you'll need to pass `&b` to `dagjson.Encode`, as it is `*bytes.Buffer` (the pointer to the type, rather than the type itself) that implements the `io.Writer` interface.